### ISSUE TITLE
TypeError: object Future can't be used in 'await' expression (resulting in 500 at login)

### ISSUE THREAD

**Issue by amit1rrr** [jupyterhub/jupyterhub#1751]
I just setup jupyterhub from the source. When I try to login  I get following error:
```
[E 2018-03-24 01:18:45.094 JupyterHub web:1621] Uncaught exception POST /hub/login?next= (127.0.0.1)
HTTPServerRequest(protocol='http', host='127.0.0.1:8081', method='POST', uri='/hub/login?next=', version='HTTP/1.1', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/Users/amitrathi/.virtualenvs/jphub3/lib/python3.6/site-packages/tornado/web.py", line 1543, in _execute
    result = yield result
  File "/workspace/jupyterhub/jupyterhub/handlers/login.py", line 81, in post
    user = await self.login_user(data)
  File "/workspace/jupyterhub/jupyterhub/handlers/base.py", line 429, in login_user
    authenticated = await self.authenticate(data)
  File "/workspace/jupyterhub/jupyterhub/auth.py", line 228, in get_authenticated_user
    authenticated = await self.authenticate(handler, data)
TypeError: object Future can't be used in 'await' expression
```

I haven't made any changes to the default config file. Running it on MacOS high sierra. 
I am trying to login with my system username & password since I haven't added any users to the db yet.

**Comment by amit1rrr** [comment]
I am trying to login with my system username & password since I haven't added any users to the db yet.

**Comment by loginoff** [comment]
I get the exact same error on login, but using the _jupyterhub/jupyterhub_ container.
This is the version inside the container
```
root@287f969c2eae:~# jupyterhub --version
0.9.0.dev
```

**Comment by minrk** [comment]
Duplicate of #1732 closed by #1749
